### PR TITLE
Tracked last message and tweaked additive accordingly using class method

### LINKED ISSUE #108: Additive text allows adding to other people's text
**Expected behavior:** When someone other than me speaks, the "Additive" text option should have no effect.

**Actual behavior:** When someone other than me speaks, I can use the "Additive" text option to add more text to their chatbox, even though I have no control over their character.

**Fix:** Track character ID of who said the last message. Ignore additive option if the last message's character ID does not match that of the incoming message.

### PR DESCRIPTION
**Pull request by caleb-mabry**
Fixes #108 

Tracks the last message send with a class variable and we use class methods to alter that variable. I've tested this and it works, feel free to checkout the branch and try it for yourself.
I'm sure the logic could be tweaked to be a little cleaner, but that file is such a mess still.

### CODE REVIEW

File: server/network/aoprotocol.py [hunk 1]
```diff
@@ -38,6 +39,7 @@ class ProtocolError(Exception):
 
 class AOProtocol(asyncio.Protocol):
     """The main class that deals with the AO protocol."""
+    last_message_char_id: int = -1
```
Comment (oldmud0) [inline server/network/aoprotocol.py:42]: This should go in the area class

**Review by oldmud0 — CHANGES_REQUESTED** [review]

File: server/network/aoprotocol.py (hunk 1 above)
Comment (caleb-mabry) [reply server/network/aoprotocol.py:42]: Gotcha. Thanks!

Comment (caleb-mabry) [reply server/network/aoprotocol.py:42]: I've added them to the area and the protocol checks to see if it's allowed @oldmud0

**Comment by caleb-mabry** [comment]
Changes have been applied, feel free to take a look

File: server/area_manager.py [hunk 2]
```diff
@@ -121,6 +121,15 @@ def remove_client(self, client: ClientManager.Client):
             if client.char_id != -1:
                 database.log_room('area.leave', client, self)
 
+        def client_can_additive(self, client: ClientManager.Client):
+            if self.last_ic_message is None:
```
Comment (oldmud0) [inline server/area_manager.py:125]: Seems like you forgot to commit something. `self.last_ic_message` should be defined in the constructor of `Area`.

**Review by oldmud0 — CHANGES_REQUESTED** [review]

File: server/area_manager.py (hunk 2 above)
Comment (caleb-mabry) [reply server/area_manager.py:125]: You angry banana. It's in there @oldmud0 image

Comment (oldmud0) [reply server/area_manager.py:125]: ...oh

**Review by oldmud0 — APPROVED** [review]

**Pull request merged**